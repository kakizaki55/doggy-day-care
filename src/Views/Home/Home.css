.dog-list {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  background-image: url('../../assests/cool-background.svg');
  background-color: rgb(228, 216, 128);
}
