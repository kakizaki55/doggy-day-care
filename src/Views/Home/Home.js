import React from 'react';
import { useEffect, useState } from 'react';
import fetchDogs from '../../services/fetchDogs';

export default function Home() {
  const [dogs, setDogs] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogs();
      setDogs(data);
    };
    fetchData();
  }, []);
  console.log(dogs);

  return <div>this is the home page</div>;
}
