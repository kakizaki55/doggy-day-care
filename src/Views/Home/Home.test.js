import Home from './Home';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
test('span shot test for the home View container', async () => {
  const { container } = await render(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );
  await screen.findByText('Meet Melba !');
  expect(container).toMatchSnapshot();
});
