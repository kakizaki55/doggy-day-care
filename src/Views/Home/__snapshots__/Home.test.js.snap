// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`span shot test for the home View container 1`] = `
<div>
  <div
    class="dog-list"
  >
    <a
      href="/dog/6"
    >
      <div
        class="dog-card"
      >
        <h1>
           Meet 
          Melba
           ! 
        </h1>
        <img
          src="https://placedog.net/500?id=8"
        />
        <p>
          Melba
           is a(n) 
          4
           year old 
          Austrailian Cattle Dog
           
        </p>
      </div>
    </a>
    <a
      href="/dog/7"
    >
      <div
        class="dog-card"
      >
        <h1>
           Meet 
          Liliana
           ! 
        </h1>
        <img
          src="https://placedog.net/500?id=5"
        />
        <p>
          Liliana
           is a(n) 
          1
           year old 
          Husky
           
        </p>
      </div>
    </a>
    <a
      href="/dog/8"
    >
      <div
        class="dog-card"
      >
        <h1>
           Meet 
          Amely
           ! 
        </h1>
        <img
          src="https://placedog.net/500?id=12"
        />
        <p>
          Amely
           is a(n) 
          0
           year old 
          Rottweiler
           
        </p>
      </div>
    </a>
    <a
      href="/dog/9"
    >
      <div
        class="dog-card"
      >
        <h1>
           Meet 
          Althea
           ! 
        </h1>
        <img
          src="https://placedog.net/500?id=10"
        />
        <p>
          Althea
           is a(n) 
          1
           year old 
          German Short-haired Pointer
           
        </p>
      </div>
    </a>
    <a
      href="/dog/10"
    >
      <div
        class="dog-card"
      >
        <h1>
           Meet 
          Franny
           ! 
        </h1>
        <img
          src="https://placedog.net/500?id=51"
        />
        <p>
          Franny
           is a(n) 
          4
           year old 
          French Bulldog
           
        </p>
      </div>
    </a>
    <a
      href="/dog/4"
    >
      <div
        class="dog-card"
      >
        <h1>
           Meet 
          Eyes days
           ! 
        </h1>
        <img
          src="https://placedog.net/500?id=22"
        />
        <p>
          Eyes days
           is a(n) 
          5
           year old 
          Weimaraner Mix
           
        </p>
      </div>
    </a>
    <a
      href="/dog/3"
    >
      <div
        class="dog-card"
      >
        <h1>
           Meet 
          Ronny Ron
           ! 
        </h1>
        <img
          src="https://images.pexels.com/photos/6246667/pexels-photo-6246667.jpeg?auto=compress&cs=tinysrgb&dpr=3&h=750&w=1260"
        />
        <p>
          Ronny Ron
           is a(n) 
          1
           year old 
          Golden Retriever
           
        </p>
      </div>
    </a>
    <a
      href="/dog/5"
    >
      <div
        class="dog-card"
      >
        <h1>
           Meet 
          Pugilicious
           ! 
        </h1>
        <img
          src="https://placedog.net/500?id=26"
        />
        <p>
          Pugilicious
           is a(n) 
          11
           year old 
          Pug
           
        </p>
      </div>
    </a>
    <a
      href="/dog/11"
    >
      <div
        class="dog-card"
      >
        <h1>
           Meet 
          PIXIEL
           ! 
        </h1>
        <img
          src="https://images.cdn1.stockunlimited.net/preview1300/pixel-dog_2021383.jpg"
        />
        <p>
          PIXIEL
           is a(n) 
          22
           year old 
          german byter 
           
        </p>
      </div>
    </a>
    <a
      href="/dog/1"
    >
      <div
        class="dog-card"
      >
        <h1>
           Meet 
          Barton the third
           ! 
        </h1>
        <img
          src="https://placedog.net/500?id=6"
        />
        <p>
          Barton the third
           is a(n) 
          3
           year old 
          Weimaraner
           
        </p>
      </div>
    </a>
    <a
      href="/dog/2"
    >
      <div
        class="dog-card"
      >
        <h1>
           Meet 
          Happy
           ! 
        </h1>
        <img
          src="https://placedog.net/500?id=21"
        />
        <p>
          Happy
           is a(n) 
          11
           year old 
          French Bulldog
           
        </p>
      </div>
    </a>
  </div>
</div>
`;
